src/board/UBBoardView.cpp

```diff
@@ -1492,29 +1494,74 @@ void UBBoardView::mouseDoubleClickEvent (QMouseEvent *event)
     mousePressEvent (event);
 }
 
+// Mouse Wheel Zoom
+// Source: https://wiki.qt.io/Smooth_Zoom_In_QGraphicsView
 void UBBoardView::wheelEvent (QWheelEvent *wheelEvent)
 {
-    QList<QGraphicsItem *> selItemsList = scene()->selectedItems();
-    // if NO have selected items, than no need process mouse wheel. just exist
-    if( selItemsList.count() > 0 )
+
+    if( wheelEvent->modifiers() & Qt::ControlModifier )
+    {
+        wheelZoomEvent(wheelEvent);
+    }
+    else
     {
-        // only one selected item possible, so we will work with first item only
-        QGraphicsItem * selItem = selItemsList[0];
 
-        // get items list under mouse cursor
-        QPointF scenePos = mapToScene(wheelEvent->pos());
-        QList<QGraphicsItem *> itemsList = scene()->items(scenePos);
 
-        bool isSelectedAndMouseHower = itemsList.contains(selItem);
-        if(isSelectedAndMouseHower)
+        QList<QGraphicsItem *> selItemsList = scene()->selectedItems();
+        // if NO have selected items, than no need process mouse wheel. just exist
+        if( selItemsList.count() > 0 )
         {
-            QGraphicsView::wheelEvent(wheelEvent);
-            wheelEvent->accept();
-        }
+            // only one selected item possible, so we will work with first item only
+            QGraphicsItem * selItem = selItemsList[0];
 
+            // get items list under mouse cursor
+            QPointF scenePos = mapToScene(wheelEvent->pos());
+            QList<QGraphicsItem *> itemsList = scene()->items(scenePos);
+
+            bool isSelectedAndMouseHower = itemsList.contains(selItem);
+            if(isSelectedAndMouseHower)
+            {
+                QGraphicsView::wheelEvent(wheelEvent);
+                wheelEvent->accept();
+            }
+
+        }
     }
+}
+
+void UBBoardView::wheelZoomEvent(QWheelEvent *event){
+    int numDegrees = event->delta() / 8;
+     int numSteps = numDegrees / 15; // see QWheelEvent documentation
+     _numScheduledScalings += numSteps;
+     if (_numScheduledScalings * numSteps < 0) // if user moved the wheel in another direction, we reset previously scheduled scalings
+     _numScheduledScalings = numSteps;
```

Comment:
Another remark on these lines: the code resets the `_numScheduledScalings` variable, but there might still be several `QTimeLine` objects running which fire their events. So if you e.g. create three wheel events in one direction and then turn back the wheel by one step, you will still end up with four `QTimeLine` instances creating their events mostly in parallel. With the current code it is not easily possible to cancel those. I wonder whether it would not be better to use a single `QTimeLine` instance and to adjust duration etc accordingly.